controllers/profiles/reconciler.go

```diff
@@ -64,8 +66,10 @@ type ProfileReconciler interface {
 
 // stateSupport is the shared structure with common accessors used throughout the whole reconciliation profiles
 type stateSupport struct {
-	logger *logr.Logger
-	client client.Client
+	logger                          *logr.Logger
+	client                          client.Client
+	recorder                        record.EventRecorder
+	numberOfBuildAttemptsAfterError int
```

Comment:
Do we really need  the `numberOfBuildAttemptsAfterError`?
Since the stateSupport has go the client we are able every time to retrieve the build and or the platform CR (we probably already doing this) and read it from them.

Reply:
Is here because in the first implementation we passed it in the state support, but since we have in the platform we can remove here.